### PR TITLE
feat(console): Service Plan - Application entitlement assignment

### LINKED ISSUE #2405: feat(console): Service Plan - Application entitlement assignment
Dependencies: #2400 for `stripe` module

Implement:
- [x] Figma `Service Plan` and `Service Plan \ Purchase` frames
- [x] Assignment of new plan should take the form of 2 persistence elements: an edge from user's AccountURN to ApplicationURN with tag `pays/app` and a property in the Starbase DO for app details with the plan name, eg. `servicePlan`: `Pro`. The "Assigned Entitlements" section of the Billing & Invoicing page should reflect this assignment.
- [x] Upgrading a plan (when an entitlement is available) should result in an unassignment of the existing plan and assignment of the new plan, in a DO transaction.
- [ ] When entitlements for a desired plan aren't available, and a application-specific purchase is initated, it should display the purchase confirmation modal that, upon confirmation, converts the Purchase button to a spinner that checks the completion state of the assignment request, periodically. On the backend, the confirmation should initiate a purchase from the `stripe` module, with the additional metadata set for the `clientId` the entitlement is being purchased for. 
- [ ] Success Stripe request processing should be modified to assign a newly purchased entitlement to the app referenced, if in the metadata of the webhook request from Stripe, the app clientId is populated. The processing should update the button/spinner accordingly.

Alternatives, in case transaction-level metadata isn't supported for the Purchase webhook:
- When receiving confirmation of purchase action from user, store the clientId in a `pendingServicePlans` key in the DO, ideally with a success transaction ID received when submitting the transaction with Stripe. Upon receiving the webhook, look up the `pendingServicePlans` for any pending plans that match the details in the webhook. When a match is found, assign entitlement to the application with matched `clientId` and remove the `clientId` from the `pendingServicePlans` structure.

### PR DESCRIPTION
**Pull request by Cosmin-Parvulescu**
### Description

- Implemented Figma designs
- Added appPlan to appDetails so that we can access the app's current plan throughout console
- Added returnURL to updatePaymentMethod in stripe client so that we can return to multiple places
- Added FREE plan to ServiceTypes as well as plans.ts
- Added `handled` metadata when creating / updating subscriptions. This can be set to whatever value and if it's present in the webhook event, it A - clears it from the subscription and B - returns without updating entitlements. This allows for sychronous payments.
- Added status pill component
- Added new edge type (`pays/app`)

### Related Issues

- Closes #2405 

### Testing

* Adding these two lines to the `nodes/account.ts` file allows one to reset their clientID as well as paymentInformation and existing service plans:
```
  async getServicePlans(): Promise<ServicePlans | undefined> {
    await this.state.storage.delete('stripePaymentData') // This 
    await this.state.storage.delete('servicePlans') // and this

    return this.state.storage.get<ServicePlans>('servicePlans')
  }
```

After that, it's a matter of going through the flows.

### Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) guidelines
- [x] I have tested my code (manually and/or automated if applicable)
- [ ] I have updated the documentation (if necessary)

### CODE REVIEW

**Review by szkl — CHANGES_REQUESTED** [review]
It'd be nice to avoid the long `switch` statement in `apps/console/app/routes/apps/$clientId/gnillib.tsx`.

File: apps/console/app/root.tsx [hunk 1]
```diff
@@ -68,29 +68,32 @@ export const meta: MetaFunction = () => ({
   viewport: 'width=device-width,initial-scale=1',
 })
 
+export type AppLoaderData = {
+  clientId: string
+  name?: string
+  icon?: string
+  published?: boolean
+  createdTimestamp?: number
```
[the remark this answers is not in the document]
Comment (Cosmin-Parvulescu) [reply apps/console/app/root.tsx:76]: Hmm, I couldn't say; just extracted a type from the initial LoaderData that can be used somewhere else.

File: apps/console/app/root.tsx [hunk 2]
```diff
@@ -68,29 +68,32 @@ export const meta: MetaFunction = () => ({
...
+    WALLET_CONNECT_PROJECT_ID: string
   }
 }
 
 export const loader: LoaderFunction = getRollupReqFunctionErrorWrapper(
   async ({ request, context }) => {
     const jwt = await requireJWT(request)
     const traceHeader = generateTraceContextHeaders(context.traceSpan)
-    const parsedJwt = parseJwt(jwt)
+    const parsedJwt = parseJwt(jwt!)
```
[the remark this answers is not in the document]
Comment (Cosmin-Parvulescu) [reply apps/console/app/root.tsx]: Added a check

File: apps/console/app/routes/__layout/gnillib/payment.tsx [hunk 3]
```diff
@@ -27,6 +27,12 @@ export const loader: LoaderFunction = getRollupReqFunctionErrorWrapper(
       accountURN,
     })
 
-    return updatePaymentMethod({ customerID })
+    const headers = request.headers
+    const returnURL = headers.get('Referer') as string
```
[the remark this answers is not in the document]
Comment (Cosmin-Parvulescu) [reply apps/console/app/routes/__layout/gnillib/payment.tsx]: Checked

**Comment by Cosmin-Parvulescu** [comment]
> It'd be nice to avoid the long `switch` statement in `apps/console/app/routes/apps/$clientId/gnillib.tsx`.

I extracted the two methods to make it a bit more readable I guess.

**Comment by szkl** [comment]
image

**Review by szkl — APPROVED** [review]

**Pull request merged**